src/main/java/org/jabref/gui/preferences/preview/PreviewTabViewModel.java

```diff
@@ -157,8 +157,10 @@ public void setPreviewLayout(PreviewLayout selectedLayout) {
             sourceTextProperty.setValue(layout.getText().replace("__NEWLINE__", "\n"));
             selectedIsEditableProperty.setValue(true);
         } else {
-            sourceTextProperty.setValue(((CitationStylePreviewLayout) selectedLayout).getSource());
-            selectedIsEditableProperty.setValue(false);
+            if (selectedLayout instanceof CitationStylePreviewLayout) {
+                sourceTextProperty.setValue(((CitationStylePreviewLayout) selectedLayout).getSource());
+                selectedIsEditableProperty.setValue(false);
+            }
         }
```

Comment:
- The case of BstLayout is missing
- Use this new instanceof pattern thing: One can directly assign the value to the variable. See lines 156 and 157 how this works. (Otherwise, line 157 would have read `...((TextBasedPreviewLeayout) layout).getText()...`

Reply:
This is about displaying the content for the source tab, but BstPreviewLaoyut doesn't yet have a method getSource, it only accepts a path. One would need to add a new method and return the file content as string? Is that an option? I have no idea how large bst files are

Reply:
like @Siedlerchr said getSource() might need to be implemented or  case of bstlayout needs to be handled separately as @koppor mentioned I've left space for that in my recent commit otherwise its causing runtime error and also PreviewLayout doesn't have the getSource() method so its type casting to Citationstylepreviewlayout was necessary I guess
> * Use this new instanceof pattern thing: One can directly assign the value to the variable. See lines 156 and 157 how this works. (Otherwise, line 157 would have read `...((TextBasedPreviewLeayout) layout).getText()...`
what is new instanceof pattern thing?